### PR TITLE
Add a config for ECS deployment

### PR DESCRIPTION
**Pull request by TaKO8Ki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref https://github.com/pipe-cd/pipe/issues/1568

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -48,6 +48,8 @@ const (
 	KindLambdaApp Kind = "LambdaApp"
 	// KindCloudRunApp represents deployment configuration for a CloudRun application.
 	KindCloudRunApp Kind = "CloudRunApp"
+	// KindEcsApp represents deployment configuration for an AWS ECS.
+	KindEcsApp Kind = "EcsApp"
```
Comment (khanhtc1202) [inline pkg/config/config.go:52]: should be `ECSApp` 🤔

Comment (TaKO8Ki) [reply pkg/config/config.go]: I replaced "EcsApp" with "ECSApp". 👍

File: pkg/config/config.go [hunk 2]
```diff
@@ -48,6 +48,8 @@ const (
 	KindLambdaApp Kind = "LambdaApp"
 	// KindCloudRunApp represents deployment configuration for a CloudRun application.
 	KindCloudRunApp Kind = "CloudRunApp"
+	// KindEcsApp represents deployment configuration for an AWS ECS.
+	KindEcsApp Kind = "ECSApp"
```
Comment (nghialv) [inline pkg/config/config.go:52]: nit: `KindECSApp`

File: pkg/config/config.go [hunk 3]
```diff
@@ -82,6 +84,7 @@ type Config struct {
 	TerraformDeploymentSpec  *TerraformDeploymentSpec
 	CloudRunDeploymentSpec   *CloudRunDeploymentSpec
 	LambdaDeploymentSpec     *LambdaDeploymentSpec
+	EcsDeploymentSpec        *EcsDeploymentSpec
```
Comment (nghialv) [inline pkg/config/config.go:87]: nit: `ECSDeploymentSpec`

File: pkg/config/deployment_ecs.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	if err := s.GenericDeploymentSpec.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+type EcsDeploymentInput struct {
+	// The name of service definition file placing in application directory.
+	// Default is servicedef.yaml
+	ServiceDefinition string `json:"serviceDefinition"`
```
Comment (nghialv) [inline pkg/config/deployment_ecs.go:37]: `ServiceDefinitionFile` and `TaskDefinitionFile` for better understand.

Comment (TaKO8Ki) [reply pkg/config/deployment_ecs.go]: I added a suffix to them.

File: pkg/config/config.go (hunk 3 above)
Comment (TaKO8Ki) [reply pkg/config/config.go]: I replaced Ecs with ECS.

**Comment by TaKO8Ki** [comment]
@nghialv 
I'm ready for your review.

File: pkg/config/testdata/application/ecs-app.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: pipecd.dev/v1beta1
+kind: ECSApp
+spec:
+  input:
+    serviceDefinition: path/to/servicedef.yaml
```
Comment (nghialv) [inline pkg/config/testdata/application/ecs-app.yaml:5]: This should be fixed too.

File: pkg/config/deployment_ecs.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	return nil
+}
+
+type ECSDeploymentInput struct {
+	// The name of service definition file placing in application directory.
+	// Default is servicedef.yaml
+	ServiceDefinitionFile string `json:"serviceDefinition"`
+	// The name of task definition file placing in application directory.
+	// Default is taskdef.yaml
+	TaskDefinitionFile string `json:"taskDefinition"`
```
Comment (nghialv) [inline pkg/config/deployment_ecs.go:40]: The tag should be fixed too.

**Review by cakecatz — APPROVED** [review]
TS file changing looks good to me.

**Comment by TaKO8Ki** [comment]
@nghialv 
I'm ready for your review again!

**Comment by nghialv** [comment]
Thank you.
/lgtm

File: pkg/config/testdata/application/ecs-app.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+apiVersion: pipecd.dev/v1beta1
+kind: ECSApp
+spec:
+  input:
+    serviceDefinitionFile: path/to/servicedef.yaml
+    taskDefinitionFile: path/to/taskdef.yaml
```
Comment (nakabonne) [inline pkg/config/testdata/application/ecs-app.yaml:6]: very very nits:
```suggestion
    serviceDefinitionFile: /path/to/servicedef.yaml
    taskDefinitionFile: /path/to/taskdef.yaml
```

Comment (nakabonne) [reply pkg/config/testdata/application/ecs-app.yaml:6]: To emphasize that it must be an absolute path, just in case.

Comment (TaKO8Ki) [reply pkg/config/testdata/application/ecs-app.yaml]: I accepted your suggestion!

**Comment by nakabonne** [comment]
Thank you!
/approve

**Pull request merged**